### PR TITLE
227 sentinel audit logs stdout grpc

### PR DESCRIPTION
**Pull request by sahinakyol**
# #227 sentinel audit logs stdout with grpc

## Description

Created Logger with grpc server client sends from safe to sentinel stdout logs. Related [comment](https://github.com/vmware-tanzu/secrets-manager/issues/227#issuecomment-1936103427).

## Changes

List the major changes you have made in bullet points:

- Makefile
- app/sentinel/busywait/main.go
- app/sentinel/internal/safe/get.go
- app/sentinel/internal/safe/post.go
- app/sentinel/logger/generated/
- app/sentinel/logger/log.proto
- app/sentinel/logger/log_builder.go
- app/sentinel/logger/log_builder_test.go
- app/sentinel/logger/logclient.go
- app/sentinel/logger/logger_port.go
- app/sentinel/logger/logger_test.go
- go.mod
- go.sum
- hack/bundle.sh
- hack/generate-proto-files.sh
- makefiles/VSecMGenerateProtoFiles.mk
- makefiles/VSecMHelp.mk

## Test Policy Compliance

- [x] I have added or updated unit tests for my changes.
- [x] I have included integration tests where applicable.
- [x] All new and existing tests pass successfully.

## Code Quality

- [x] I have followed the coding standards for this project.
- [x] I have performed a self-review of my code.
- [x] My code is well-commented, particularly in areas that may be difficult 
      to understand.

## Documentation

- [x] I have made corresponding changes to the documentation (if applicable).
- [x] I have updated any relevant READMEs or wiki pages.

## Additional Comments

Include any additional comments or context about the PR here.

## Checklist

Before you submit this PR, please make sure:
- [x] You have read [the contributing guidelines][contributing] and 
      *especially* the [test policy][test-policy].
- [x] You have thoroughly tested your changes.
- [x] You have followed all the contributing guidelines for this project.
- [x] You understand and agree that your contributions will be publicly available 
  under the project’s license.

[contributing]: https://vsecm.com/docs/contributing/
[test-policy]: https://vsecm.com/docs/contributing/#add-tests-for-new-features

*By submitting this pull request, you confirm that my contribution is made under 
the terms of the project’s license and that you have the authority to grant 
these rights.*

---

Thank you for your contribution to [**VMware Secrets Manager**](https://vsecm.com)
🐢⚡️!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,3 +59,6 @@ include makefiles/Git.mk
 
 ## Coverage
 include makefiles/Test.mk
+
+## Generate Proto Files
+include makefiles/VSecMGenerateProtoFiles.mk
```
Comment (v0lkan) [inline Makefile:64]: We might need to add documentation for this.

File: app/sentinel/logger/log_builder.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package logger
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+var currentTime = func() string {
```
Comment (v0lkan) [inline app/sentinel/logger/log_builder.go:9]: why not

```go
func currentTime() string {
	return time.Now().Local().Format(time.DateTime)
}
```

**Comment by v0lkan** [comment]
`make build-local` failed with the following reason:


```
=> ERROR [builder 8/8] RUN CGO_ENABLED=0 GOOS=linux go build -mod vendor -a -o vsecm-safe ./app/safe/cmd/main.go                                                                0.2s
------
 > [builder 8/8] RUN CGO_ENABLED=0 GOOS=linux go build -mod vendor -a -o vsecm-safe ./app/safe/cmd/main.go:
0.200 go: inconsistent vendoring in /build:
0.200   google.golang.org/protobuf@v1.32.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
0.200   google.golang.org/protobuf@v1.30.0: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod
0.200
0.200   To ignore the vendor directory, use -mod=readonly or -mod=mod.
0.200   To sync the vendor directory, run:
0.200           go mod vendor
```

**Comment by v0lkan** [comment]
Reverting the change.

**Comment by v0lkan** [comment]
cc: @sahinakyol .

File: app/sentinel/logger/log_builder.go (hunk 2 above)
Comment (sahinakyol) [reply app/sentinel/logger/log_builder.go:9]: For testing

**Pull request merged**